submission/programs/adae.R

```diff
@@ -0,0 +1,283 @@
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC02FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER == "Y"
+  ) %>%
+  #---------------------------------------------------------------------#
+  # AOCC03FL - 1st Occurrence 03 Flag for Serious SOC                   #
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID, AEBODSYS),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC03FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER == "Y"
+  ) %>%
+  #---------------------------------------------------------------------#
+  # AOCC04FL - 1st Occurrence 04 Flag for Serious PT                    #
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID, AEBODSYS, AEDECOD),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC04FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER == "Y"
+  ) %>%
+  # --------------------------------------------------------------------#
+  # CQ01NAM - Customized Query 01 Name                                  #
+  # --------------------------------------------------------------------#
+  mutate(CQ01NAM = ifelse(str_detect(AEDECOD, "APPLICATION") |
+    str_detect(AEDECOD, "DERMATITIS") |
+    str_detect(AEDECOD, "ERYTHEMA") |
+    str_detect(AEDECOD, "BLISTER") |
+    str_detect(AEBODSYS, "SKIN AND SUBCUTANEOUS TISSUE DISORDERS") &
+      !str_detect(AEDECOD, "COLD SWEAT|HYPERHIDROSIS|ALOPECIA"),
+  "DERMATOLOGIC EVENTS",
+  NA_character_
+  )) %>%
+  # --------------------------------------------------------------------#
+  # AOCC01FL - 1st Occurrence 01 Flag for CQ01                          #
+  # --------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      order = vars(ASTDT, AESEQ),
+      new_var = AOCC01FL,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & CQ01NAM == "DERMATOLOGIC EVENTS"
+  )
+
+#---------------------- #
+# ADAE derivation end   #
+#---------------------- #
+
+# --------------------------------------------- #
+# Check variables against define &              #
+# Assign dataset labels, var labels and formats #
+# --------------------------------------------- #
+ADAE <- adae0 %>%
+  drop_unspec_vars(adae_spec) %>% # Check all variables specified are present and no more
+  check_ct_data(adae_spec, na_acceptable = TRUE) %>% # Checks all variables with CT only contain values within the CT
```

Comment:
can we explain ct? Are those code lists?

Reply:
Controlled terminology.

Reply:
@SHAESEN2 or @thomas-neitmann If ok, please `Resolve conversation`.